pkg/providers/aws/cluster_network_provider.go

```diff
@@ -685,6 +686,63 @@ func (n *NetworkProvider) IsEnabled(ctx context.Context) (bool, error) {
 	return len(validBundledVPCSubnets) == 0, nil
 }
 
+// DeleteBundledCloudResources returns an error on any error deleting of the following resources
+// * elasticache subnet group
+// * rds subnet group
+// * ec2 security group
+//
+// it has been located under the cluster network provider as it requires 3 different aws sessions
+// (elasticache, rds and ec2) to delete the required resources even though it deals with bundled
+// resources. The majority of the functionality in this file relates to standalone aws vpc and it's
+// resources.
+func (n *NetworkProvider) DeleteBundledCloudResources(ctx context.Context) error {
+	logger := n.Logger.WithField("action", "deleteBundledCloudResources")
+
+	subnetGroupName, err := BuildInfraName(ctx, n.Client, "subnetgroup", DefaultAwsIdentifierLength)
+	if err != nil {
+		return errorUtil.Wrap(err, "error building bundle subnet group resource name on deletion")
+	}
+	logger.Info("deleting bundled elasticcache subnet group %s , if it's not found it's already deleted and will continue", subnetGroupName)
+	_, err = n.ElasticacheApi.DeleteCacheSubnetGroup(&elasticache.DeleteCacheSubnetGroupInput{
+		CacheSubnetGroupName: aws.String(subnetGroupName),
+	})
+	if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() != "CacheSubnetGroupNotFoundFault" {
```

Comment:
I believe there are variables that we can use instead of inline strings for AWS errors e.g. `elasticache.ErrCodeCacheSubnetGroupNotFoundFault`

Not super important, but it'll help us handle if that exact string ever changes.